tools/codegen/gen.py

```diff
@@ -790,6 +794,12 @@ def compute_registration_declarations(f: NativeFunction, backend_indices: Dict[D
 def _read_template(template_fn: str) -> CodeTemplate:
     return CodeTemplate.from_file(template_fn)
 
+
+# String hash that's stable across different executions, unlike builtin hash
+def string_stable_hash(s: str) -> int:
```

Comment:
This is going to reshuffle all the functions that were previously in given files right?
Did you check that the file length are still balanced in practice?

Reply:
No hash function is perfect, but I think a cryptographic has is going to be more reliable than adding all the ASCII characters together, like was done previously.

Here are the real-world distributions:

```
Sharding Operators.cpp, splits are:
 [427, 434, 459, 438, 381]
[341/1504] Generating ../../torch/csrc/autograd...esting/_internal/generated/annotated_fn_args.py
Sharding python_functions.cpp, splits are:
 [96, 101, 107, 118, 98]
Sharding python_torch_functions.cpp, splits are:
 [232, 267, 245]
Sharding VariableType.cpp, splits are:
 [251, 250, 283, 280, 237]
Sharding TraceType.cpp, splits are:
 [410, 422, 456, 461, 376]
```

Reply:
Looks good! Thanks for the numbers!